### DISCUSSION TITLE
[Idea] Multicarrier scenario > Split Order Vs Single Order

### DISCUSSION THREAD

**Discussion by MateoCastilloPS** [PrestaShop/PrestaShop#36798] [Ideas]
### The problem

Hello all, 

As you all well know there are many historic issues related to multi-carrier management in prestashop ([summarised in this EPIC](https://github.com/PrestaShop/PrestaShop/issues/15491)). 

We are currently thinking on solutions to avoid these pains for future prestashop versions, thinking about the best possible approach for users in the future. 

### The solution

One of the approaches would be to drop the order splitting in multicarrier scenario, and create instead a new logic in which the « splitting » would happen at a fulfilment level **inside** the order. 

Our hypothesis is that this approach could :  
* Improve the merchant experience when he is managing his orders (1 order id, 1 order ref)
* Improve the end customer experience as he would have one order reference everywhere
* Avoid complex calculations when it comes to splitting orders with discounts, refunds etc… 
* Ease up integrating with external platforms (ERP, WMS, TMS, OMS…. Etc)
* Ease up payment module integrations to avoid issues related to transactions 

Of course this approach would have many technical implications at many levels, for now we are at the start of the solution phase. 

### Alternatives

The alternative approach would be to keep the split order concept and keep the logic of 1 order = 1 carrier, rebuild properly the Front office layouts to avoid issues listed [in the EPIC](https://github.com/PrestaShop/PrestaShop/issues/15491), as well as spend time patching all the splitting calculations issues. 

This would also need to be sized in effort and we yet need to discover the implications of correcting all the existing issues properly.

### Additional context

So today we come to you to ask you for your help on gathering insights, based on your expertise, and from your client’s experiences: 

- which of these two concepts (single order vs split order) do you think would be best for all users and why ? (Users = dev, merchant, end customers) 
- how did your customers react when they discovered that the orders were split ?  Do they like it ? Do they understand it ? How are they impacted by it ? 
- taking aside all the known issues regarding multicarrier, if you take the concept of it, to split an order, how does it impact your developments and integrations with third party services ? 

On our end in the product team we are also preparing to start conducting some interviews, I share to you all [my calendly](https://meetings.hubspot.com/mateo-castillo) in here if you wish to share your thoughts with me :) (I can run the ITW in English, French and Spanish) 

Thanks !! 

Mateo

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by web-cooking-factory** [comment]
https://github.com/PrestaShop/PrestaShop/discussions/31596

**Comment by Hlavtox** [comment]
Ability to have multiple packages per one order is one of our main pain points in Prestashop. It's super common to send order in multiple packages, and with the current system, it's impossible.

- Cannot manage shipping in 2 or more phases.
- Can't keep track of previous shipping number because it's overwritten.
- You don't know what was already sent and what wasn't.

**Comment by jouino16** [comment]
Thanks for the detailed discussion and for introducing the new shipments-based approach in PrestaShop 9.1.

As a module developer, I've been working since 2019 on a production-ready multi-carrier solution focused on the front-office side (multi-carrier selection within a single order), now available on [PrestaShop Addons](https://addons.prestashop.com/en/shipping-pickup-solutions/93768-multi-shipping-multiple-carriers-order.html).

Seeing this use case addressed at the core level confirms how recurrent and important it is for merchants, based on real-world projects over the past few years.

I'll be following the evolution of this feature in upcoming versions, especially regarding front-office behavior.
